File: Src/CSharpier.Rider/src/main/java/com/intellij/csharpier/FunctionRunner.java
package com.intellij.csharpier;

import java.util.function.Supplier;

public class FunctionRunner {
    public static String runUntilNonNull(Supplier<String>... functions) {
        for (Supplier<String> function : functions) {
            String result = function.get();
            if (result != null) {
                return result;
            }
        }
        return null;
    }
}

File: Src/CSharpier.VisualStudio/CSharpier.VisualStudio/FunctionRunner.cs
using System;

namespace CSharpier.VisualStudio
{
    using System.Linq;

    public static class FunctionRunner
    {
        public static string? RunUntilNonNull(params Func<string?>[] functions)
        {
            return functions
                .Select(possibleFunction => possibleFunction())
                .OfType<string>()
                .FirstOrDefault();
        }
    }
}
